Arriba/Arriba.Test/CustomColumn.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using Arriba.Model;
+using System.Collections.Generic;
+using Arriba.Model.Expressions;
+using Arriba.Structures;
+using Arriba.Serialization;
+
+namespace Arriba.Test
+{
+    public static class CustomColumnSupport
+    {
+        public static void RegisterCustomColumns()
+        {
+            ColumnFactory.AddColumnCreator("color", (details, columnComponents, initialCapacity) =>
```

Comment:
This is not a problem because only the built-in types get Indexed;Sorted added to their column stacks.  Column components for custom columns are passed through and not modified by the ColumnFactory.  

It is possible we may want to figure out a way to allow a custom column to just prepend/append the custom IColumn to a normally created stack but that is an API for another iteration.